internal/services/network/application_gateway_resource.go

```diff
@@ -4801,6 +4803,10 @@ func applicationGatewayCustomizeDiff(ctx context.Context, d *pluginsdk.ResourceD
 		}
 	}
 
+	if oldTier != tier && (strings.EqualFold(tier, string(applicationgateways.ApplicationGatewayTierBasic)) || strings.EqualFold(tier, string(applicationgateways.ApplicationGatewayTierStandard)) || strings.EqualFold(tier, string(applicationgateways.ApplicationGatewayTierWAF))) {
+		return fmt.Errorf("the creation of new %q V1 SKU is no longer supported, please use V2 SKU with \"Basic\", \"Standard_v2\" or \"WAF_v2\" tier instead", tier)
+	}
+
```

Comment:
This error message is misleading, this is not always a new creation. And also there is no V2 SKU for Basic.

Recommended error msg: `New creation / update to %q SKU tier is no longer supported, please use supported SKU tiers: ...`